addons/purchase_stock/models/purchase_order_line.py

```diff
@@ -342,6 +342,10 @@ def _prepare_purchase_order_line_from_procurement(self, product_id, product_qty,
         if po.partner_id.group_rfq == 'week' and po.partner_id.group_on != 'default':
             delta_days = (7 + int(po.partner_id.group_on) - res['date_planned'].isoweekday()) % 7
             res['date_planned'] = fields.Datetime.to_datetime(res['date_planned']) + relativedelta(days=delta_days)
+            if not po.date_planned or po.date_planned >= res['date_planned']:
```

Comment:
If the `date_planned` moved, shouldn't we shift the `date_order` as well, regardless of if the new `date_planned` is smaller than the old one?